spec/unit/pdk/util_spec.rb

```diff
@@ -0,0 +1,242 @@
+require 'spec_helper'
+
+describe PDK::Util do
+  shared_context :with_version_file, version_file: true do
+    let(:version_file) { File.join('path', 'to', 'the', 'version', 'file') }
+
+    before(:each) do
+      allow(PDK::Util::Version).to receive(:version_file).and_return(version_file)
+    end
+  end
+
+  shared_context :without_version_file, version_file: false do
+    before(:each) do
+      allow(PDK::Util::Version).to receive(:version_file).and_return(nil)
+    end
+  end
+
+  describe '.find_upwards' do
+    subject { described_class.find_upwards(target, start_dir) }
+
+    let(:target) { 'metadata.json' }
+    let(:start_dir) { nil }
+    let(:actual_start_dir) { '/path/to/something/deep/in/a/module' }
+
+    before(:each) do
+      allow(File).to receive(:directory?).with('/path/to/something/deep/in/a/module').and_return(true)
```

Comment:
I added a commit that unrolled the loop here so we just have a dumb list of expected messages; to keep the unit test code as simple as possible.